src/components/Forms/Health/HTTPHealthFormEditor.tsx

```diff
@@ -0,0 +1,94 @@
+import FormikEnvVarConfigsFields from "../Formik/FormikConfigEnvVarFields";
+import FormikConfigFormFieldsArray from "../Formik/FormikConfigFormFieldsArray";
+import FormikTextInput from "../Formik/FormikTextInput";
+import FormikCheckbox from "../Formik/FormikCheckbox";
+import FormikTemplateFields from "../Formik/FormikTemplateFields";
+import FormikJSONCheckFields from "../Formik/FormikJSONCheckFields";
+import FormikCheckboxFieldsGroup from "../Formik/FormikCheckboxFieldsGroup";
+
+type HTTPHealthFormEditorProps = {
+  fieldName: string;
+};
+
+export function HTTPHealthFormEditor({ fieldName }: HTTPHealthFormEditorProps) {
+  return (
+    <>
+      <FormikTextInput name={`${fieldName}.name`} label="Name" required />
+      <FormikTextInput name={`${fieldName}.description`} label="Description" />
+      <FormikTextInput name={`${fieldName}.icon`} label="Icon" />
+      {/* add labels here */}
+      <FormikTemplateFields name={`${fieldName}.test`} label="Test" />
+      <FormikTemplateFields name={`${fieldName}.display`} label="Display" />
+      <FormikTemplateFields name={`${fieldName}.transform`} label="Transform" />
+      <FormikTextInput name={`${fieldName}.endpoint`} label="Endpoint" />
+      <FormikTextInput name={`${fieldName}.namespace`} label="Namespace" />
+      <FormikTextInput
+        name={`${fieldName}.thresholdMillis`}
+        label="Threshold Millis"
+      />
+      <FormikConfigFormFieldsArray
+        fields={[
+          {
+            fieldComponent: FormikTextInput,
+            name: `${fieldName}.responseCodes`
+          }
+        ]}
+        label="Response Codes"
+        name={`${fieldName}.responseCodes`}
+      />
+      <FormikTextInput
+        name={`${fieldName}.responseContent`}
+        label="Response Content"
+      />
+      <FormikJSONCheckFields
+        label="Response JSON Checks"
+        name={`${fieldName}.responseJSONChecks`}
+      />
+      <FormikTextInput
+        name={`${fieldName}.maxSSLExpiry`}
+        label="Max SSL Expiry"
+      />
+      <FormikTextInput name={`${fieldName}.method`} label="Method" />
+      <FormikTextInput name={`${fieldName}.ntlm`} label="ntlm" />
```

Comment:
Use a button group:
Authentication:  None | Basic | NTLM | NTLMv2
and then have username/password below which gating them - If you specify a auth method, you must specify username and password